Content.Server/SimpleStation14/Jukebox/JukeboxSystem.cs

```diff
@@ -0,0 +1,308 @@
+    /// </summary>
+    public void UnPause(EntityUid jukeBox, JukeboxComponent jukeboxComp)
+    {
+        if (!jukeboxComp.CanPlay)
+            return;
+
+        if (!jukeboxComp.Paused)
+            return;
+
+        jukeboxComp.Paused = false;
+
+        TryRestart(jukeBox, jukeboxComp);
+    }
+
+    /// <summary>
+    ///     Plays a song in the Jukebox using the provided <see cref="JukeboxTrackPrototype"/> and optional offset.
+    ///     If the Jukebox is already playing a song, it will be stopped and replaced with the new song.
+    ///     To queue a song instead, see <see cref="TryQueueSong"/>.
+    /// </summary>
+    /// <param name="song">The JukeboxTrackPrototype representing the song to be played.</param>
+    /// <param name="offset">The optional offset from the start of the song to begin playing from.</param>
+    public void Play(EntityUid jukeBox, JukeboxComponent jukeboxComp, string song, TimeSpan offset = new TimeSpan())
+    {
+        if (!_prototype.TryIndex<JukeboxTrackPrototype>(song, out var songPrototype))
+        {
+            Logger.Error($"Jukebox track prototype {song} not found!");
+
+            Skip(jukeBox, jukeboxComp);
+            return;
+        }
+
+        Play(jukeBox, jukeboxComp, songPrototype, offset);
+    }
+
+    /// <inheritdoc cref="Play(EntityUid, JukeboxComponent, string, TimeSpan)"/>
+    /// <remarks>
+    ///     Directly takes a <see cref="JukeboxTrackPrototype"/> instead.
+    /// </remarks>
+    public void Play(EntityUid jukeBox, JukeboxComponent jukeboxComp, JukeboxTrackPrototype song, TimeSpan offset = new TimeSpan())
+    {
+        if (!jukeboxComp.CanPlay)
+            return;
+
+        Clean(jukeBox, jukeboxComp);
+
+        jukeboxComp.Paused = false;
+
+        jukeboxComp.Playing = true;
+
+        jukeboxComp.CurrentlyPlayingTrack = song;
+
+        jukeboxComp.FinishPlayingTime = _timing.CurTime + song.Duration - offset;
+
+        jukeboxComp.CurrentlyPlayingStream = _audio.Play(song.Path, Filter.Broadcast(), jukeBox, true, AudioParams.Default.WithPlayOffset((float)offset.TotalSeconds));
+
+        Timer.Spawn((int) (song.Duration.TotalMilliseconds - offset.TotalMilliseconds), () => OnSongEnd(jukeBox, jukeboxComp), jukeboxComp.SongTimerCancel.Token);
+
+        UpdateState(jukeBox, jukeboxComp);
+    }
+
+    #endregion Public functions
+    #region Private functions
+
+    /// <summary>
+    ///     Sets the Jukebox to either a state where it can or cannot play songs.
+    /// </summary>
+    private void CheckCanPlay(EntityUid uid, JukeboxComponent component)
+    {
+        var canPlay = true;
+
+        if (EntityManager.TryGetComponent<ApcPowerReceiverComponent>(uid, out var powerComp) && !powerComp.Powered)
+            canPlay = false;
+
+        component.CanPlay = canPlay;
+
+        if (canPlay)
+            TryRestart(uid, component);
+        else
+            Stop(uid, component);
+    }
```

Comment:
Should probably be public, honestly? I dunno, maybe if we added a way to state that Jukeboxes were disabled or something.

OH speaking of which EMPs need to do something. I'm thinking- would it be too much to stop the song, play it with the correct offset from where it was but add like random reverb to it or some shit? We can check the AudioParams, I think they got some funny stuff. After EMP duration, just stop it again and play like normal.